### ISSUE TITLE
[ban-types] message for `Object` is confusing

### ISSUE THREAD

**Issue by six-transit** [typescript-eslint/typescript-eslint#5191]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYewdgzgLgBA7gCwIZQKYDdUCcBcMDyARgFarCwC8MA5AgJbUDcQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AIwEMnajTUMkRNGgB7aJHBgAviGlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const whatever: Object = 'hi';
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/ban-types": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

The message should not claim that using `Object` as a type is "marginally better than `unknown`." This is arguably false, very confusing -- especially for newcomers to TypeScript -- and contradictory, considering the message subsequently suggests... using `unknown`.

One suggestion would be to simply remove that part of the sentence:

> Don't use `Object` as a type. The `Object` type actually means "any non-nullish value."

This would [align it with the simpler warning message for `{}`](https://github.com/typescript-eslint/typescript-eslint/blob/2a345b40508c61917b0488f688c94407f79cb7f2/packages/eslint-plugin/src/rules/ban-types.ts#L84-L99).

### Actual Result

Error message:

> Don't use `Object` as a type. The `Object` type actually means "any non-nullish value", so it is marginally better than `unknown`.
> - If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
> - If you want a type meaning "any value", you probably want `unknown` instead. 1:17 - 1:23

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.28.0` |
| `@typescript-eslint/parser`            | `5.28.0` |
| `TypeScript`                           | `4.7.3` |
| `ESLint`                               | `8.17.0` |
| `node`                                 | `18.3.0` |

**Comment by bradzacher** [comment]
> The message should not claim that using Object as a type is "marginally better than unknown." This is arguably false

What's arguably false about it?

As the message states - `Object` accepts anything, [except for `null` or `undefined`](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXyjDBAAcMBnAeQCMArcDAClgHMAueG+zASg4DccWYAG4AUGID0kgkVIYQwMYWJkqdBowCMPcSvnruTAORRju5XLVdNGGMhAX91jZkYBvAL5OrFG24BtAF0fVT9XJkYeeABeAD54L1CDfyZIKHJyRO89X0NNVBAAd3gAZRAmHgspGVQcDAIICBwixUsw-Lc0UEQsQuBklyNGVGQm3SA).

When compared to `unknown` - `Object` is only differentiated by it not allowing `null`/`undefined`.

There's little actual benefit to restricting `null`/`undefined` at a location considering that you're going to have to refine your way out of the type to do anything with it. This means that there's not actually any additional safety gained from the restriction.

One important point of consideration is that `unknown` is well understood by all TS users to mean "accepts anything, but you need to refine out of it". On the contrary - `Object` *looks* like it should only accept objects - it's very misleading unless you really understand TS's "structural typing" model. I've seen many TS newbies use it thinking such, and thus introduced a large holes into the types.

So by all accounts - "marginally better than unknown" is pretty apt, IMO.

---

> very confusing -- especially for newcomers to TypeScript

As mentioned, from my experience teaching people TS - `Object` is a very confusing type for people. Which is the point of the expanded wording - to really highlight that it's not a good type to use.

> and contradictory, considering the message subsequently suggests... using unknown.

I'm not sure I understand what's contradictory about suggesting using the clearer type, when (in our opinion) it's just better to use `unknown`.

---

> This would [align it with the simpler warning message for {}](https://github.com/typescript-eslint/typescript-eslint/blob/2a345b40508c61917b0488f688c94407f79cb7f2/packages/eslint-plugin/src/rules/ban-types.ts#L84-L99).

The reason that `{}` has a different message is because of the intention when using the type.
When people write `{}` they almost always want a type which means "an object with no properties". But that's not what they get.

OTOH when people use `Object` they usually mean "any object, I don't care about its shape".

The wording is different because the intent is different. Note that `{}` also has an additional suggestion in its message suggesting how one might write an empty object type.

**Comment by bradzacher** [comment]
> It can easily be read to suggest that Object is in fact a good type to use. After all, it's better than unknown.

I would agree with that if the sentence was entirely standalone, but the message starts with "Don't use Object as a type.".
IMO the ambiguity only stands when taken out of context.

**Comment by Josh-Cena** [comment]
The problem is that "marginally better" is a very subjective judgment. A type isn't better because it's stricter—only because it describes the intent. `Object` is not a good type because it almost never catches the user intent. In that sense, it's much much worse than `unknown`, which everyone knows what its semantics are.

**Comment by bradzacher** [comment]
> but if you think they're stupid or wrong that's your call

To be clear - I don't think anyone here is stupid.

---

I think that the wording is good-enough as is, and I don't think it's worth-while to bikeshed a one word change that only marginally alters the meaning.
If there is a larger change we want to make to rewrite the message - I'd be open to it.
